src/api/MainApi.ts

```diff
@@ -201,22 +207,53 @@ export default class MainApi {
      * }
      * ```
      * */
-    public getArticles = async (
+    public getArticlesFromDaysAgoTillDate = async (
         retryType: RetryType = RetryType.RETRY_INDEFINETELY,
         maxRetries = DEFAULT_MAX_RETRIES,
         waitingTime = 3,
+        days: number,
+        end: string,
         retryCount = 0
     ) => {
```

Comment:
i parametri obbligatori dovrebbero sempre stare prima dei parametri opzionali, non ha senso dare un default a un parametro se ci sono parametri successivi senza default, non c'è alcun modo di usare il valore di default così.

in generale come best practice i parametri obbligatori devono sempre essere i primi
```suggestion
    public getArticlesFromDaysAgoTillDate = async (
        days: number,
        end: string,
        retryType: RetryType = RetryType.RETRY_INDEFINETELY,
        maxRetries = DEFAULT_MAX_RETRIES,
        waitingTime = 3,
        retryCount = 0
    ) => {
```

potenzialmente però dato che sono sempre le stesse impostazioni, potresti pensare di wrapparle in un oggetto così che chi chiama la funzione può modificare anche solo uno dei campi, senza modificarli necessariamente tutti in ordine (se voglio modificare `waitingTime` devo mettere prima `retryType` e `maxRetries` necessariamente)

ma ancora, dato che passi questi valori pari pari alla funzione `istance.get` di axios, magari avrebbe addirittura senso prendere un'intero oggetto `AxiosRequestConfig`? su quest'ultima opzione non sono troppo sicuro ahah, però vedi te, alla fine della fiera voglio poter fare
```ts
api.getArticlesFromDaysAgoTillDate(7, new Date().toISOString())
```
senza dover ripassare tutti i soliti valori di default che tanto verranno messi quasi sempre uguali